builder/eth_service.go

```diff
@@ -0,0 +1,66 @@
+package builder
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/beacon"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/eth"
+	"github.com/ethereum/go-ethereum/eth/catalyst"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type IEthereumService interface {
+	BuildBlock(attrs *BuilderPayloadAttributes) *beacon.ExecutableDataV1
+	GetBlockByHash(hash common.Hash) *types.Block
+	Synced() bool
+}
+
+type testEthereumService struct {
+	synced             bool
+	testExecutableData *beacon.ExecutableDataV1
+	testBlock          *types.Block
+}
+
+func (t *testEthereumService) BuildBlock(attrs *BuilderPayloadAttributes) *beacon.ExecutableDataV1 {
+	return t.testExecutableData
+}
+
+func (t *testEthereumService) GetBlockByHash(hash common.Hash) *types.Block { return t.testBlock }
+
+func (t *testEthereumService) Synced() bool { return t.synced }
+
+type EthereumService struct {
+	eth *eth.Ethereum
+}
+
+func NewEthereumService(eth *eth.Ethereum) *EthereumService {
+	return &EthereumService{eth: eth}
+}
+
+func (s *EthereumService) BuildBlock(attrs *BuilderPayloadAttributes) *beacon.ExecutableDataV1 {
+	// Create an empty block first which can be used as a fallback
+	empty, err := s.eth.Miner().GetSealingBlockSync(attrs.HeadHash, uint64(attrs.Timestamp), attrs.SuggestedFeeRecipient, attrs.GasLimit, attrs.Random, true)
```

Comment:
This one is not needed in our non-FCU case (builds an empty block to calculate payload id)